implementations/rust/ockam/ockam_api/src/session/sessions.rs

```diff
@@ -1,18 +1,24 @@
-use core::any::Any;
 use core::fmt;
 use core::future::Future;
 use core::pin::Pin;
+use std::time::Duration;
+
 use minicbor::bytes::ByteArray;
 use minicbor::{Decode, Encode};
+use tracing as log;
+
 use ockam_core::compat::collections::HashMap;
 use ockam_core::compat::rand;
-use ockam_core::Error;
-use ockam_multiaddr::MultiAddr;
-use std::sync::{Arc, Mutex};
-use tracing as log;
+use ockam_core::{Error, Route};
 
-pub type Replacement = Pin<Box<dyn Future<Output = Result<MultiAddr, Error>> + Send>>;
-pub type Replacer = Box<dyn FnMut(MultiAddr) -> Replacement + Send>;
+//most sessions replacer are dependent on the node manager, if many session
```

Comment:
Off-topic: do you think instead of having lock for a NodeManager state we should have individual locks for its fields?

Reply:
Splitting for each field is most likely an overkill, but we can definitively split it up and separate read-only from writable fields. I think most usages are actually derived from some kind of registry and don't even need an async lock, a sync lock or an atomic update will probably suffice.